packages/@aws-cdk/core/lib/cfn-resource.ts

```diff
@@ -289,6 +293,45 @@ export class CfnResource extends CfnRefElement {
     addDependency(this, target, `"${Node.of(this).path}" depends on "${Node.of(target).path}"`);
   }
 
+  /**
+   * Indicates that this resource no longer depends on another resource.
+   *
+   * This can be used for resources across stacks (including nested stacks)
+   * and the dependency will automatically be removed from the relevant scope.
+   */
+  public removeDependsOn(target: CfnResource) : void {
+    // skip this dependency if the target is not part of the output
+    if (!target.shouldSynthesize()) {
+      return;
+    }
+
+    removeDependency(this, target, `"${Node.of(this).path}" depends on "${Node.of(target).path}"`);
+  }
+
+  /**
+   * Retrieves an array of resources this resource depends on.
+   *
+   * This assembles dependencies on resources across stacks (including nested stacks)
+   * automatically.
+   */
+  public obtainDependsOn() : Element[] {
```

Comment:
I'm not so sure about this naming. Can this be refined a bit more? I think I'd be unsure if it was obtaining the resources that depend on this one or if it's obtaining the resources that depend on it.

Reply:
Ohhh. Now I see that an existing addDependsOn functions is probably why these were named this way. Interestingly, I see people not really using that. I wonder if it's because of the odd phrasing of it. Maybe we deprecate that naming convention and use a new one? What do you think?

Reply:
Yeah, I Like `*Dependency` much better as well. I will adjust.

Reply:
So the only reason I can think of to avoid the `addDependency` name in particular is potential confusion with `Node.addDependency()`. Maybe matching the name makes sense, since the higher-level Construct has the higher level interface with a matching name then, but maybe that could also be confusing. What do you think?